transcendence-app/src/chat/chat.controller.ts

```diff
@@ -261,6 +261,28 @@ export class ChatController {
     return updatedChatroom;
   }
 
+  @Delete('room/:chatroomId')
+  @ApiForbiddenResponse({ description: 'Forbidden' })
+  @ApiNotFoundResponse({ description: 'Not Found' })
+  @ApiServiceUnavailableResponse({ description: 'Service unavailabe' })
+  async deleteChatroom(
+    @GetUser() userMe: User,
+    @Param('chatroomId', ParseUUIDPipe) chatroomId: string,
+  ): Promise<Chatroom> {
+    const chatroom = await this.chatService.getChatroomById(chatroomId);
+    if (!chatroom) {
+      throw new NotFoundException();
+    }
+    if (userMe.id !== chatroom.ownerId) {
+      throw new ForbiddenException();
+    }
+    const deletedChatroom = await this.chatService.deleteChatroom(chatroomId);
+    if (!deletedChatroom) {
+      throw new ServiceUnavailableException();
```

Comment:
Not sure if we should go with `ServiceUnavailableException()` or `UnprocessableEntityException()`

Reply:
thought: For this case, `ServiceUnavailableException()` is better because we first check if there is a chatroom with this ID, and this query should succeed unless the database server is down or malfunctioning.